clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/controllers/EcsFindImagesByTagController.java

```diff
@@ -76,42 +77,46 @@ private NetflixAmazonCredentials getCredentials(String accountId) {
 
   @RequestMapping(value = "/images/find", method = RequestMethod.GET)
   public Object findImage(@RequestParam("q") String dockerImageUrl, HttpServletRequest request) {
-    String accountId = dockerImageUrl.split("\\.")[0];
-    String tag = dockerImageUrl.split("\\:")[1];
-    String repository = dockerImageUrl.split("\\/")[1].split(":")[0]; //TODO - these regexes make my eyes bleed.  Use a better approach.
+    Matcher uriRegexMatcher = REPOSITORY_URI_PATTERN.matcher(dockerImageUrl);
+    if (!uriRegexMatcher.find()) {
+      throw new Error("The repository URI provided is not proper.");
+    }
+
+    String accountId = uriRegexMatcher.group(1);
```

Comment:
You could use this code block to better instruct the error message @dkirillov

Reply:
Done.